interpreter/subtype_check.gen.go

```diff
@@ -0,0 +1,417 @@
+			}
+
+			return false
+		}
+
+		// An intersection type `T{Us}`
+		// is a subtype of an intersection type `V{Ws}`:
+		switch typedSubType := subType.(type) {
+		case *IntersectionStaticType:
+
+			// When `T == AnyResource || T == AnyStruct || T == Any`:
+			// not statically.
+			switch typedSubType.LegacyType {
+			case nil,
+				PrimitiveStaticTypeAny,
+				PrimitiveStaticTypeAnyStruct,
+				PrimitiveStaticTypeAnyResource:
+				return false
+			}
+
+			switch typedSubTypeLegacyType := typedSubType.LegacyType.(type) {
+			case *CompositeStaticType:
+
+				// When `T != AnyResource && T != AnyStructType && T != Any`: if `T == V`.
+				// `Us` and `Ws` do *not* have to be subsets:
+				// The owner may freely restrict and unrestrict.
+				return typedSubTypeLegacyType == typedSuperType.LegacyType
+			}
+
+			return false
+		case *CompositeStaticType:
+			return IsSubType(typeConverter, typedSubType, typedSuperType.LegacyType)
+		}
+
+		// A type `T`
+		// is a subtype of an intersection type `AnyResource{Vs}` / `AnyStruct{Vs}` / `Any{Vs}`:
+		// not statically.
+		switch subType {
+		case PrimitiveStaticTypeAny,
+			PrimitiveStaticTypeAnyStruct,
+			PrimitiveStaticTypeAnyResource:
+			return false
+		}
+
+		return IsParameterizedSubType(typeConverter, subType, typedSuperType)
+
+	case FunctionStaticType:
+		switch typedSubType := subType.(type) {
+		case FunctionStaticType:
+
+			// View functions are subtypes of impure functions
+			switch typedSubType.Purity {
+			case typedSuperType.Purity,
+				FunctionPurityView:
+
+				// Type parameters must be equivalent. This is because for subtyping of functions,
+				// parameters must be *contravariant/supertypes*, whereas, return types must be *covariant/subtypes*.
+				// Since type parameters can be used in both parameters and return types, inorder to satisfies both above
+				// conditions, bound type of type parameters can only be strictly equal, but not subtypes/supertypes of one another.
+				typedSubTypeTypeParameters := typedSubType.TypeParameters
+				typedSuperTypeTypeParameters := typedSuperType.TypeParameters
+				if len(typedSubTypeTypeParameters) != len(typedSuperTypeTypeParameters) {
+					return false
+				}
+
+				for i, source := range typedSubTypeTypeParameters {
+					target := typedSuperTypeTypeParameters[i]
+					if !(deepEquals(source.TypeBound, target.TypeBound)) {
+						return false
+					}
+				}
+
+				// Functions are contravariant in their parameter types.
+				typedSubTypeParameters := typedSubType.Parameters
+				typedSuperTypeParameters := typedSuperType.Parameters
+				if len(typedSubTypeParameters) != len(typedSuperTypeParameters) {
+					return false
+				}
+
+				for i, source := range typedSubTypeParameters {
+					target := typedSuperTypeParameters[i]
+					if
+
+					// Note the super-type is the subtype's parameter
+					// because the parameters are contravariant.
+					!(sema.IsSubType(target.TypeAnnotation.Type, source.TypeAnnotation.Type)) {
```

Comment:
Same here, the `FunctionStaticType` is just a wrapper around `sema.FunctionType`. Therefore the parameter type (i.e: `target.TypeAnnotation.Type`) is a `sema.Type` (not a `StaticType`). So calling `IsSubType` method of the interpreter package (like every other place inside this method) isn't going to work. Needed to call the `IsSubType` method from the `sema` package. So the customizer will update here as well, to the desire code, similarly to the previous case.